pkg/api/handlers/libpod/images.go

```diff
@@ -607,70 +608,29 @@ func SearchImages(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	filter := image.SearchFilter{}
-	if len(query.Filters) > 0 {
-		if len(query.Filters["stars"]) > 0 {
-			stars, err := strconv.Atoi(query.Filters["stars"][0])
-			if err != nil {
-				utils.InternalServerError(w, err)
-				return
-			}
-			filter.Stars = stars
-		}
-		if len(query.Filters["is-official"]) > 0 {
-			isOfficial, err := strconv.ParseBool(query.Filters["is-official"][0])
-			if err != nil {
-				utils.InternalServerError(w, err)
-				return
-			}
-			filter.IsOfficial = types.NewOptionalBool(isOfficial)
-		}
-		if len(query.Filters["is-automated"]) > 0 {
-			isAutomated, err := strconv.ParseBool(query.Filters["is-automated"][0])
-			if err != nil {
-				utils.InternalServerError(w, err)
-				return
-			}
-			filter.IsAutomated = types.NewOptionalBool(isAutomated)
-		}
+	_, authfile, key, err := auth.GetCredentials(r)
+	if err != nil {
+		utils.Error(w, "failed to retrieve repository credentials", http.StatusBadRequest, errors.Wrapf(err, "failed to parse %q header for %s", key, r.URL.String()))
+		return
 	}
-	options := image.SearchOptions{
+	defer auth.RemoveAuthfile(authfile)
+
+	options := entities.ImageSearchOptions{
+		Authfile: authfile,
 		Limit:    query.Limit,
 		NoTrunc:  query.NoTrunc,
 		ListTags: query.ListTags,
-		Filter:   filter,
+		Filters:  query.Filters,
 	}
-
 	if _, found := r.URL.Query()["tlsVerify"]; found {
-		options.InsecureSkipTLSVerify = types.NewOptionalBool(!query.TLSVerify)
+		options.SkipTLSVerify = types.NewOptionalBool(!query.TLSVerify)
 	}
-
-	_, authfile, key, err := auth.GetCredentials(r)
+	ir := abi.ImageEngine{Libpod: runtime}
+	reports, err := ir.Search(r.Context(), query.Term, options)
 	if err != nil {
-		utils.Error(w, "failed to retrieve repository credentials", http.StatusBadRequest, errors.Wrapf(err, "failed to parse %q header for %s", key, r.URL.String()))
+		utils.Error(w, "Something went wrong.", http.StatusNotFound, err)
```

Comment:
We need to distinguish between Not Found and Internal Server Error here.

Reply:
NotFound is not an error, it just returns nothing.
```
$ ./bin/podman search registry.access.redhat.com/danwalsh
$ ./bin/podman-remote  search registry.access.redhat.com/danwalsh
$
```

Reply:
If NotFound is not an error, why are we returning it here? Docker API indicates the correct code is 500 - https://docs.docker.com/engine/api/v1.40/#operation/ImageSearch

Reply:
Well podman search imagenotfound 

Does not return an error.

I don't think we should do podman search imagenotfound to return an error.

But if we have to, then I will need to do this for only compat endpoint.